src/query/query_executor.c

```diff
@@ -7261,156 +7265,130 @@ qexec_execute_scan (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl
 
   do
     {
+    try_scan_next_scan:
+
       sc_scan = scan_next_scan (thread_p, &xasl->curr_spec->s_id);
       if (sc_scan != S_SUCCESS)
 	{
 	  return sc_scan;
 	}
 
-      /* set scan item as qualified */
-      qualified = true;
-
-      if (qualified)
+      /* clear bptr subquery list files */
+      if (xasl->bptr_list)
 	{
-	  /* clear bptr subquery list files */
-	  if (xasl->bptr_list)
-	    {
-	      qexec_clear_head_lists (thread_p, xasl->bptr_list);
-	      if (xasl->curr_spec->flags & ACCESS_SPEC_FLAG_FOR_UPDATE)
-		{
-		  scan_operation_type = S_UPDATE;
-		}
-	      else
-		{
-		  scan_operation_type = S_SELECT;
-		}
-	    }
+	  qexec_clear_head_lists (thread_p, xasl->bptr_list);
+	  scan_operation_type = (xasl->curr_spec->flags & ACCESS_SPEC_FLAG_FOR_UPDATE) ? S_UPDATE : S_SELECT;
+
 	  /* evaluate bptr list */
-	  for (xptr = xasl->bptr_list; qualified && xptr != NULL; xptr = xptr->next)
+	  for (xptr = xasl->bptr_list; xptr != NULL; xptr = xptr->next)
 	    {
 	      if (qexec_execute_obj_fetch (thread_p, xptr, xasl_state, scan_operation_type) != NO_ERROR)
 		{
 		  return S_ERROR;
 		}
 	      else if (xptr->proc.fetch.fetch_res == false)
 		{
-		  qualified = false;
+		  goto try_scan_next_scan;
```

Comment:
I disagree that using the goto statement is better for readability.

Reply:
In general, goto statements reduce readability.
But wouldn't this be an efficient method for sources that are long due to many repeated and nested if() blocks?